lib/fluent/plugin/in_opensearch.rb

```diff
@@ -302,6 +340,9 @@ def run
           run_slice(slice_id)
         end
       end
+    rescue Faraday::ConnectionFailed => e
+      log.warn "Connection to OpenSearch failed during search in the 'run' method: #{e.message}. Retrying..."
+      retry
```

Comment:
Infinite retrying without any conditions are sick for us. We should provide some of the choices whether this feature should be enabled or not. What you you think?